### ISSUE TITLE
15$: Parse Git merge conflict markers

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#404]
Example:
```
<<<<<<< HEAD
      let parser;
      try {
        // TODO(strager): Reuse processes across documents.
        let process = await this._documentProcessManager.getOrCreateProcessAsync();
        parser = await process.createDocumentForVSCodeAsync();
      } catch (e) {
        if (e instanceof ProcessCrashed) {
          throw new LintingCrashed(e);
        } else {
          throw e;
        }
      }
||||||| parent of f1942618 (@@@)
      let factory = await this._documentProcessManager._processFactoryPromise;
      // TODO(strager): Reuse processes across documents.
      let process = await factory.createProcessAsync();
      let parser = await process.createDocumentForVSCodeAsync();
=======
      // TODO(strager): Reuse processes across documents.
      let parser = await process.createDocumentForVSCodeAsync();
>>>>>>> f1942618 (@@@)
```

**Comment by singalhimanshu** [comment]
What errors should be reported?
Currently `quick-lint-js` reports the following errors:
```sh
quick-lint-js main.js
main.js:1:1: error: missing operand for operator [E026]
main.js:1:1: error: missing operand for operator [E026]
main.js:1:3: error: missing operand for operator [E026]
main.js:1:5: error: missing operand for operator [E026]
main.js:14:1: error: missing operand for operator [E026]
main.js:14:1: error: missing operand for operator [E026]
main.js:14:3: error: missing operand for operator [E026]
main.js:14:5: error: missing operand for operator [E026]
main.js:14:15: error: missing semicolon after statement [E027]
main.js:14:18: error: missing semicolon after statement [E027]
main.js:14:29: error: unexpected '@' [E042]
main.js:14:30: error: unexpected '@' [E042]
main.js:14:31: error: unexpected '@' [E042]
main.js:18:11: error: redeclaration of variable: parser [E034]
main.js:2:11: note: variable already declared here [E034]
main.js:19:1: error: missing operand for operator [E026]
main.js:19:1: error: missing operand for operator [E026]
main.js:19:4: error: missing operand for operator [E026]
main.js:19:1: error: invalid expression left of assignment [E020]
main.js:21:10: error: missing semicolon after statement [E027]
main.js:22:1: error: missing operand for operator [E026]
main.js:22:1: error: missing operand for operator [E026]
main.js:22:4: error: missing operand for operator [E026]
main.js:22:19: error: unexpected '@' [E042]
main.js:22:20: error: unexpected '@' [E042]
main.js:22:21: error: unexpected '@' [E042]
main.js:1:9: warning: use of undeclared variable: HEAD [E057]
main.js:8:26: warning: use of undeclared variable: ProcessCrashed [E057]
main.js:9:21: warning: use of undeclared variable: LintingCrashed [E057]
main.js:14:16: warning: use of undeclared variable: of [E057]
main.js:14:19: warning: use of undeclared variable: f1942618 [E057]
main.js:21:7: warning: use of undeclared variable: let [E057]
main.js:22:9: warning: use of undeclared variable: f1942618 [E057]
```

**Comment by vegerot** [comment]
the most basic thing:
```
error: merge conflict marker detected
```

Maybe get fancier and try linting the conflicted code (maybe by disabling all lints that require context from outside the line).  But to start, you could just report the error

**Comment by strager** [comment]
I think error recovery would be too complicated. I think a conflict marker should end parsing.

**Comment by EmptyGroceryBag** [comment]
> No, because that'd cause problems
Clearly you don't think my idea is good. However, this is a _linter_. It's the programmer's job to write code that works. What do you think this is, Copilot?
